src/platform/intel/cavs/platform.c

```diff
@@ -543,7 +543,47 @@ void platform_wait_for_interrupt(int level)
 }
 #endif
 
+#define ADSP_IMR_MAGIC_VALUE		0x02468ACE
+#define IMR_L1_CACHE_ADDRESS		0xB0000000
+
+struct imr_header {
+	uint32_t adsp_imr_magic;
+	uint32_t structure_version;
+	uint32_t structure_size;
+	uint32_t imr_state;
+	uint32_t imr_size;
+	void *imr_restore_vector;
+};
+
+struct imr_state {
+	struct imr_header header;
+	uint8_t reserved[0x1000 - sizeof(struct imr_header)];
```

Comment:
PAGE_SIZE - sizeof(header) ?

Reply:
> PAGE_SIZE - sizeof(header) ?

We don't have a PAGE concept and no PAGE_SIZE macro neither?

The structs are defined in the ROM, we just following it here.

Reply:
ok, lets have a comment that these structure are immutable as the are used by the ROM.

Reply:
> ping - these will need comments that these are ROM structures and are immutable.

comments are in line 553 and line 579 already.